pkg/router/controller/route_secret_manager_test.go

```diff
@@ -718,6 +760,210 @@ func TestRouteSecretManager(t *testing.T) {
 			expectedEventType: watch.Modified,
 		},
 
+		// scenarios when route is updated (old route with externalCertificate, new route with different externalCertificate)
+		{
+			name: "route updated: old route with externalCertificate, new route with different externalCertificate denied",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "different-tls-secret",
+						},
+					},
+				},
+			},
+			secretManager: fake.SecretManager{
+				Secret: fakeSecret("sandbox", "different-tls-secret", corev1.SecretTypeTLS, map[string][]byte{
+					"tls.crt": []byte("my-crt"),
+					"tls.key": []byte("my-key"),
+				}),
+				IsPresent:  true,
+				SecretName: "tls-secret", // Used by LookupRouteSecret() to get the old secretName
+			},
+			allow:     false,
+			eventType: watch.Modified,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "different-tls-secret",
+						},
+					},
+				},
+			},
+			expectedEventType: watch.Deleted,
+			expectedRejections: []string{
+				"sandbox-route-test:ExternalCertificateValidationFailed",
+			},
+			expectedError: true,
+		},
+		{
+			name: "route updated: old route with externalCertificate, new route with different externalCertificate allowed but secret not found",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "different-tls-secret",
+						},
+					},
+				},
+			},
+			secretManager: fake.SecretManager{
+				Secret: fakeSecret("other-sandbox", "different-tls-secret", corev1.SecretTypeTLS, map[string][]byte{
+					"tls.crt": []byte("my-crt"),
+					"tls.key": []byte("my-key"),
+				}),
+				IsPresent:  true,
+				SecretName: "tls-secret", // Used by LookupRouteSecret() to get the old secretName
```

Comment:
Since the `TestRouteSecretManager` test has grown, I think keeping some comments for certain test cases always helps in easier understanding and future debugging. WDYT?

Reply:
I'm fine with it.